### ISSUE TITLE
Bug (Godot 4.3 beta 1): Can't move items in inventory, or from inventory to ItemSlot.

### ISSUE THREAD

**Issue by LeeWannacott** [peter-kish/gloot#215]
Moving inventory items results in a crash.

Steps to reproduce: Drag and drop inventory item to a new position inside itself, or a ItemSlot/ other inventory.
![image](https://github.com/peter-kish/gloot/assets/49783296/5ca35111-10be-4fb2-a63c-af1c0f0feb0c)

I think you can check for false instead of null now; from patch notes (https://godotengine.org/article/dev-snapshot-godot-4-3-beta-1/):

Breaking changes:
* Freed objects are now different than null in comparison operators, and evaluate as falsy ([GH-73896](https://github.com/godotengine/godot/pull/73896)).

**Comment by peter-kish** [comment]
Yeah checking for `false` would fix the issue in 4.3 beta 1, but I'm afraid it would introduce potential issues in 4.2. So I just went with the `is_instance_valid` check, which should work in both versions.
